test/html/document.js

```diff
@@ -88,3 +88,10 @@ const issue187 = (new DOMParser).parseFromString(`<!DOCTYPE html><html><head></h
 issue187.body.innerHTML = '<span></span>';
 let span = issue187.body.firstElementChild;
 assert(span.ownerDocument.body, issue187.body);
+
+// namespaces
+const issue189Div = document.createElementNS("http://www.w3.org/1999/xhtml", "DIV")
+assert(issue189Div.namespaceURI, 'http://www.w3.org/1999/xhtml', '<div> element, when created with createElementNS() must keep its namespace')
```

Comment:
if this is the reason for the whole dance then I'd like to introduce an `XHTML` variant among `XML` and `SVG` and move the namespace as static getter through the prototype in all cases ... this makes the code identical as it is and just enrich edge cases instead of penalizing with an extra field all elements ... what do you think?

Reply:
Actually ... never mind, this *is* the namespaceURI for HTML: https://developer.mozilla.org/en-US/docs/Web/API/Document/createElementNS#important_namespace_uris

And here I've done this test:

```js
customElements.define('a-li', class extends HTMLLIElement{}, {extends: 'li'});
document.createElement('li', {is: 'a-li'}) instanceof customElements.get('a-li'); // true
document.createElementNS('http://www.w3.org/1999/xhtml', 'li', {is: 'a-li'}) instanceof customElements.get('a-li'); // also true
```

So this PR breaks custom elements builtin extends on `createElementNS` with valid HTML namespace, hence it's a no-go to me as it is. Let's move `namespaceURI` in respective class prototypes instead, so that no performance regression can really happen, and nothing would break with builtin extends.

Reply:
P.S. dare I say this test should be part of the coverage too, so these issues would be clear out of the box.